code/obj/storage/gang_crate.dm

```diff
@@ -167,67 +172,196 @@
+			trapped = FALSE
+			toggle_tracking(FALSE)
+			icon_state = "gang_dufflebag_trap"
+			cant_self_remove = TRUE
+			cant_drop = TRUE
+			var/area/area = get_area(src)
+			playsound(src.loc, 'sound/impact_sounds/Generic_Snap_1.ogg', 50, 1)
+			boutput(user, SPAN_ALERT("As you pick up \the [src.name], a series of barbs emerge from the handle, lodging in your hand!"))
+			src.owning_gang.broadcast_to_gang("The bag [src.informant] knew about has just been stolen! Looks like it was in \the [area.name]")
+			ON_COOLDOWN(src,"bleed_msg", 30 SECONDS) //set a 30 second timer to remind players to remove this
+			idiot.setStatus("gang_trap", duration = INFINITE_STATUS)
+			H.emote("scream")
+			H.bleeding = max(1,H.bleeding)
+			SPAWN(1 SECOND)
+				bleed_process()
+
+	handle_other_remove(mob/source, mob/living/carbon/human/target)
+		if (!cant_drop || !idiot || !source.get_gang())
+			return ..()
+
+		source.visible_message("[source] rips the [src] right off [target]! Ouch!","You rip the duffle bag from [target]'s hand.")
+		idiot.emote("scream")
+		playsound(idiot.loc, 'sound/impact_sounds/Generic_Snap_1.ogg', 50, 1)
+		blood_slash(idiot, 2)
+		unhook()
+		return TRUE
+
+	proc/toggle_tracking(is_tracking)
+		if (src.tracking == is_tracking)
+			return
+
+		src.tracking = is_tracking
+
+		if (src.tracking)
+			RegisterSignal(src, XSIG_MOVABLE_AREA_CHANGED, PROC_REF(alert_gang))
+		else
+			UnregisterSignal(src, XSIG_MOVABLE_AREA_CHANGED)
+
+	proc/alert_gang(datum/component/component, area/old_area, area/new_area)
+		src.owning_gang.broadcast_to_gang("The bag [src.informant] knew about is being moved! Looks like it's been moved to \the [new_area.name]")
+		toggle_tracking(FALSE)
+
+	proc/unhook()
+		if (!idiot)
+			return
+		if (istype(idiot.l_hand, /obj/item/gang_loot) && istype(idiot.r_hand, /obj/item/gang_loot)) //this is REALLY stretching it, bub.
+			var/obj/item/gang_loot/left_loot = idiot.l_hand //like, if you hit this use case, you're something else. stealing TWO bags at once.
+			var/obj/item/gang_loot/right_loot = idiot.r_hand
+			if (!(left_loot.idiot && right_loot.idiot)) // if only one trapped bag exists, it means we're untrapping the last one
+				idiot.delStatus("gang_trap")
+		else
+			idiot.delStatus("gang_trap")
+		icon_state = "gang_dufflebag"
+		cant_self_remove = FALSE
+		cant_drop = FALSE
+		idiot = null
+
+	dropped()
+		unhook()
+		..()
+
+	proc/attempt_unhook(mob/user)
+		if (user == idiot)
+			actions.start(new /datum/action/bar/icon/unhook_gangbag(user, src),user)
+
+	attack_hand(mob/user)
+		if (user == idiot)
+			attempt_unhook(user)
+		else
+			..()
+
+	proc/bleed_process()
+		while(cant_drop)
+			if (!ON_COOLDOWN(src,"bleed_msg", 30 SECONDS))
+				boutput(idiot, SPAN_ALERT("The hooks in the bag are digging into your hands! You should pluck it out..."))
+			bleed(idiot, 1, 1)//technically doubling bleed. but it looks nice as the loops dont sync perfectly.
+			idiot.bleeding = max(1,idiot.bleeding)
+			sleep(2 SECONDS)
```

Comment:
This should be achieved through the use of the process controller.
See `proc/process` and `processing_items`.